api/src/ginger_calls.rs

```diff
@@ -684,6 +684,95 @@ pub fn get_lazy_ginger_smt_path(tree: &mut LazyGingerSMT, leaf_position: u64) ->
+}
+
+pub fn flush_bvt(tree: &mut BitvectorTree) {
+    flush_ginger_smt(tree)
+}
+
+pub fn set_bvt_persistency(tree: &mut BitvectorTree, persistency: bool) {
+    set_ginger_smt_persistency(tree, persistency)
+}
+
+pub fn get_bvt_root(tree: &BitvectorTree) -> FieldElement {
+    get_ginger_smt_root(tree)
+}
+
+pub fn get_bvt_path(tree: &mut BitvectorTree, bit_position: u64) -> GingerMHTPath {
+    get_ginger_smt_path(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn get_bvt_leaf(tree: &BitvectorTree, bit_position: u64) -> Option<FieldElement> {
+    get_leaf_in_bvt(tree, bit_position / bits_per_bvt_leaf())
+}
+
+pub fn set_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, true)
+}
+
+pub fn reset_bvt_bit(tree: &mut BitvectorTree, bit_position: u64){
+    modify_bit_in_bvt(tree, bit_position, false)
+}
+
+pub fn get_bvt_bit(tree: &BitvectorTree, bit_position: u64) -> bool {
+    let leaf_bitsize = bits_per_bvt_leaf();
+
+    let leaf_index = bit_position / leaf_bitsize;
+    if let Some(leaf) = get_leaf_in_bvt(tree, leaf_index) {
+        let leaf_bits= leaf.write_bits();
+        assert_eq!(leaf_bitsize, (leaf_bits.len() - 1) as u64);
+        leaf_bits[(leaf_bitsize - (bit_position % leaf_bitsize)) as usize]
+    } else { // leaf doesn't even exist, so the bit isn't set
+        false
+    }
+}
+
+fn bits_per_bvt_leaf() -> u64 {
+    (FieldElement::size_in_bits() - 1) as u64 // using one bit lesser to make leafs never exceed the field modulus
+}
+
+fn get_leaf_in_bvt(tree: &BitvectorTree, leaf_position: u64) -> Option<FieldElement> {
+    tree.get_leaf(Coord::new(0, leaf_position as usize))
+}
+
+fn modify_bit_in_bvt(tree: &mut BitvectorTree, bit_position: u64, bit_value: bool){
+    use algebra::{Field, FromBits};
+
+    let leaf_bitsize = bits_per_bvt_leaf();
+    let leaf_index = bit_position / leaf_bitsize;
+    // println!("leaf_bitsize: {}", leaf_bitsize);
+    // println!("leaf_index: {}", leaf_index);
+    // println!("bit_position: {}", bit_position);
+    let updated_leaf = FieldElement::read_bits(
+        {
+            let mut leaf_bits =
+                if let Some(leaf) = get_leaf_in_bvt(tree, leaf_index) {
+                    leaf.write_bits()
+                } else { // leaf doesn't exist, so create an empty leaf
+                    FieldElement::zero().write_bits()
+                };
+            assert_eq!(leaf_bitsize, (leaf_bits.len() - 1) as u64);
+            // The LSB bits are at the MSB positions
+            leaf_bits[(leaf_bitsize - (bit_position % leaf_bitsize)) as usize] = bit_value;
+            leaf_bits
+        }
+    ).unwrap();
+    // println!("updated_leaf: {}", updated_leaf);
+    if !updated_leaf.is_zero() {
+        // Write an updated leaf value
+        add_leaf_to_ginger_smt(tree, &updated_leaf, leaf_index)
+    } else {
+        // Don't store empty leafs to optimize SMT size
+        remove_leaf_from_ginger_smt(tree, leaf_index)
+    }
+}
+
```

Comment:
I think it's still needed a function that returns the whole bitvector, something like:

fn get_bitvector(tree: &BitVectorTree) -> Vec\<bool\>

Probably there might be troubles implementing this function with the implementation you provided:
1) You need a way to keep track of the biggest bit index so that you know where to stop when serializing the last leaf;
2) If for some reason you unset one or more bits of a leaf, and all the bits of that leaf will be unset, then in line 772 you are going to remove that leaf and you won't be able anymore to return it (but it should be present because that utxos exist). There shouldn't be any reason to unset a bit of the bitvector, except at each epoch start, when you reset all the bits of the bitvector (but I think you won't do it with this function), so maybe this is a false problem.

Reply:
Regarding deletion of the leafs when they don't contain any set bit: I mean that the absent leaf means that the corresponding chunk of bits is all zero. Initially the whole BVT consists of emty leafs that corresponds to all zero bits in BVT. So there is no need to store zeroized leafs. Also this will reduce size of the underlying DB. 
But actulally I also don't see the scenario where some bit should be reset except for the case, when some bit was set unintentionally and need to be unset without re-initializing whole BVT.

About serialization of the BVT - I think it is enough to get all non-empty leafs together with their positions, and then put them back in the same positions during deserialization. All empty leafs are assumed "to contain" only zero-bits, so no need to serialize them, because we already know that all empty leafs "contain" zeros.

And regarding the fn get_bitvector(tree: &BitVectorTree) -> Vec<bool>: I think it may be not possible to return all bits from BVT because based of 64-bit size of bits indexes there are at maximum 2^64 bits, so the vector to be returned is 2^61 Bytes = 2^31 GB in size, that is too much. But for a smaller trees it can be feasible. 
But actually any bit can be retrieved just by calling get_bvt_bit. So the need for get_bitvector method is a bit questionable from my viewpoint.

Reply:
It's not clear to me what will be the usage of the whole bitvector, if needed at all. But in case it's needed, I think we must serialize also the zero leaves, because they correspond to actual existing UTXOs (plus an empty leaf can be in the middle of two non empty ones). Maybe we can ask @i-Alex about this